lib/axon/compiler.ex

```diff
@@ -626,25 +636,25 @@ defmodule Axon.Compiler do
           params
           |> Enum.map(fn shape ->
             shape = apply(shape, parent_shapes)
-            apply_initializer(initializer, shape, dtype)
+            apply_initializer(initializer, shape, dtype, key)
           end)
           |> List.to_tuple()
 
         shape ->
           shape = apply(shape, parent_shapes)
-          apply_initializer(initializer, shape, dtype)
+          apply_initializer(initializer, shape, dtype, key)
       end
 
     Map.put(layer_params, name, fun)
   end
 
-  defp apply_initializer(initializer, shape, type) when is_atom(initializer) do
+  defp apply_initializer(initializer, shape, type, key) when is_atom(initializer) do
     fun = apply(Axon.Initializers, initializer, [])
-    fun.(shape, type)
+    fun.(shape, type, key)
   end
 
-  defp apply_initializer(initializer, shape, type) when is_function(initializer, 2) do
-    initializer.(shape, type)
+  defp apply_initializer(initializer, shape, type, key) when is_function(initializer, 3) do
+    initializer.(shape, type, key)
```

Comment:
Shouldn't we have to eventually split the keys? Otherwise all layers will generate the same random numbers. Perhaps support both arity 2 and arity 3 initializers. The one with arity 3 expect a key, which you split before hand?